requirements.txt

```diff
@@ -1,12 +1,53 @@
-pandas==1.3.5
-numpy==1.21.5
-numpy-financial==1.0.0
-plotly==5.8.0
-dash==2.3.1
+asttokens==2.0.5
+backcall==0.2.0
+Brotli==1.0.9
+click==8.1.3
+cx-Freeze==6.11.1
+cx-Logging==3.0
+dash==2.5.1
+dash-ace==0.2.1
+dash-bootstrap-components==1.2.0
 dash-core-components==2.0.0
 dash-html-components==2.0.0
-dash-bootstrap-components==1.1.0
+dash-table==5.0.0
+debug==0.3.2
+decorator==5.1.1
+devtools==0.8.0
+executing==0.8.3
+Flask==2.1.2
+Flask-Compress==1.12
+ipdb==0.13.9
+ipython==8.4.0
+itsdangerous==2.1.2
+jedi==0.18.1
+Jinja2==3.1.2
+MarkupSafe==2.1.1
+matplotlib-inline==0.1.3
+numpy==1.23.1
+numpy-financial==1.0.0
+packaging==21.3
+pandas==1.4.3
+parso==0.8.3
+patchelf==0.14.5.0
+pexpect==4.8.0
+pickleshare==0.7.5
+plotly==5.9.0
+prompt-toolkit==3.0.30
+ptyprocess==0.7.0
+pure-eval==0.2.2
+pydantic==1.9.1
+Pygments==2.12.0
+pyparsing==3.0.9
+python-dateutil==2.8.2
+pytz==2022.1
+see==1.4.1
+six==1.16.0
+stack-data==0.3.0
+tenacity==8.0.1
+toml==0.10.2
 tqdm==4.64.0
-cx_Freeze==6.10
-cx_Logging==3.0
-waitress==2.1.1
\ No newline at end of file
+traitlets==5.3.0
+typing_extensions==4.3.0
+waitress==2.1.2
+wcwidth==0.2.5
+Werkzeug==2.1.2
```

Comment:
typing_inspect is missing

Reply:
Resolved in https://github.com/Plutoberth/MillionSuns/pull/1/commits/48b45c7289bb2ec7b7ca23ba4a1528fecb0369f6